apps/web/app/(roles)/vaga/[id]/RolePage.tsx

```diff
@@ -0,0 +1,285 @@
+'use client'
+
+import React, { useEffect, useState, useCallback, useMemo } from 'react'
+import { FocusBanner } from 'app/landing-page/FocusBanner'
+import SanitizedHTML from 'app/components/SanitizedHTML'
+import {
+  ExternalLink,
+  MapPin,
+  AlarmClock,
+  Laptop,
+  Link,
+  Check,
+  DollarSign,
+  Calendar,
+  Share2,
+} from 'lucide-react'
+
+const COPY_TIMEOUT = 2000
+
+export const RolePage = ({ vaga }) => {
+  const [showShareMenu, setShowShareMenu] = useState(false)
+  const [isCopied, setIsCopied] = useState(false)
+  const shareMenuRef = React.useRef(null)
+
+  const isEnglish = useMemo(() => vaga.language === 'English', [vaga.language])
+
+  const shareButtonText = useMemo(
+    () =>
+      isCopied
+        ? isEnglish
+          ? 'Copied'
+          : 'Copiado'
+        : isEnglish
+        ? 'Copy link'
+        : 'Copiar link',
+    [isCopied, isEnglish]
+  )
+
+  const formatDescription = useCallback((description) => {
+    if (!description) return []
+
+    const insertLineBreaks = (text, maxLength) => {
+      const sentences = text.match(/[^.!?]+[.!?]+/g) || []
+      const result = []
+      let currentLine = ''
+
+      sentences.forEach((sentence) => {
+        if ((currentLine + sentence).length > maxLength && currentLine) {
+          result.push(currentLine.trim())
+          currentLine = ''
+        }
+        currentLine += sentence
+      })
+
+      if (currentLine) {
+        result.push(currentLine.trim())
+      }
+
+      return result
+    }
+
+    const formattedDescription = insertLineBreaks(description, 300)
+
+    return formattedDescription.map((line, index) => (
+      <p key={index} className="mb-4 whitespace-pre-line">
+        {line}
+      </p>
+    ))
+  }, [])
+
+  const isHtml = useCallback((text) => /<\/?[a-z][\s\S]*>/i.test(text), [])
+
+  const formatDate = useCallback((dateString, language) => {
+    const months = {
+      en: [
+        'Jan',
+        'Feb',
+        'Mar',
+        'Apr',
+        'May',
+        'Jun',
+        'Jul',
+        'Aug',
+        'Sep',
+        'Oct',
+        'Nov',
+        'Dec',
+      ],
+      pt: [
+        'Jan',
+        'Fev',
+        'Mar',
+        'Abr',
+        'Mai',
+        'Jun',
+        'Jul',
+        'Ago',
+        'Set',
+        'Out',
+        'Nov',
+        'Dez',
+      ],
+    }
+
+    const date = new Date(dateString)
+    const month = months[language === 'English' ? 'en' : 'pt'][date.getMonth()]
+    const day = date.getDate()
+    const year = date.getFullYear()
+
+    return `${day} ${month}, ${year}`
+  }, [])
+
+  const handleApply = useCallback(() => {
+    window.open(
+      `https://router.trampardecasa.com.br/api/role-access?roleId=${vaga.id}`,
```

Comment:
This URL is being used in different places and packages of the project, please create a function on the `shared` project named `trackedRoleURL(roleID: string)` and mutate the other projects that are using this function hardcoded.